### ISSUE TITLE
Bulk item import not working

### ISSUE THREAD

**Issue by jeremyboggs** [scholarslab/Neatline#271]
Bulk item import on current develop branch isn't working. Here's a stack trace from the errors.log in Omeka:

``` txt
2013-12-11T09:55:39-05:00 ERR (3): exception 'Zend_Controller_Router_Exception' with message 'Route id is not defined' in /Users/jeremy/Sites/omeka/application/libraries/Zend/Controller/Router/Rewrite.php:318
Stack trace:
#0 /Users/jeremy/Sites/omeka/application/libraries/Zend/Controller/Router/Rewrite.php(469): Zend_Controller_Router_Rewrite->getRoute('id')
#1 /Users/jeremy/Sites/omeka/application/libraries/Zend/View/Helper/Url.php(49): Zend_Controller_Router_Rewrite->assemble(Array, 'id', false, true)
#2 /Users/jeremy/Sites/omeka/application/views/helpers/Url.php(89): Zend_View_Helper_Url->url(Array, 'id', false, true)
#3 [internal function]: Omeka_View_Helper_Url->url(Array, 'id', Array)
#4 /Users/jeremy/Sites/omeka/application/libraries/Zend/View/Abstract.php(350): call_user_func_array(Array, Array)
#5 /Users/jeremy/Sites/omeka/application/views/helpers/RecordUrl.php(69): Zend_View_Abstract->__call('url', Array)
#6 /Users/jeremy/Sites/omeka/application/views/helpers/RecordUrl.php(69): Omeka_View->url(Array, 'id', Array)
#7 [internal function]: Omeka_View_Helper_RecordUrl->recordUrl(Object(Item), 'show', false, Array)
#8 /Users/jeremy/Sites/omeka/application/libraries/Zend/View/Abstract.php(350): call_user_func_array(Array, Array)
#9 /Users/jeremy/Sites/omeka/application/libraries/globals.php(3137): Zend_View_Abstract->__call('recordUrl', Array)
#10 /Users/jeremy/Sites/omeka/application/libraries/globals.php(3137): Omeka_View->recordUrl(Object(Item), 'show', false, Array)
#11 /Users/jeremy/Sites/omeka/application/libraries/globals.php(2251): record_url(Object(Item), 'show', false, Array)
#12 /Users/jeremy/Sites/omeka/plugins/Neatline/views/shared/exhibits/item.php(26): link_to(Object(Item), 'show', 'View the item i...')
#13 /Users/jeremy/Sites/omeka/application/libraries/Omeka/View.php(117): include('/Users/jeremy/S...')
#14 /Users/jeremy/Sites/omeka/application/libraries/Zend/View/Abstract.php(888): Omeka_View->_run('/Users/jeremy/S...')
#15 /Users/jeremy/Sites/omeka/plugins/Neatline/helpers/Views.php(55): Zend_View_Abstract->render('exhibits/item.p...')
#16 /Users/jeremy/Sites/omeka/plugins/Neatline/models/NeatlineRecord.php(295): nl_getItemMarkup(Object(NeatlineRecord))
#17 /Users/jeremy/Sites/omeka/plugins/Neatline/models/NeatlineRecord.php(307): NeatlineRecord->compileItem()
#18 /Users/jeremy/Sites/omeka/plugins/Neatline/jobs/Neatline_Job_ImportItems.php(46): NeatlineRecord->save()
#19 /Users/jeremy/Sites/omeka/plugins/Neatline/jobs/abstract/Neatline_Job_MockView.php(24): Neatline_Job_ImportItems->execute()
#20 /Users/jeremy/Sites/omeka/application/libraries/Omeka/Job/Process/Wrapper.php(29): Neatline_Job_MockView->perform()
#21 /Users/jeremy/Sites/omeka/application/scripts/background.php(61): Omeka_Job_Process_Wrapper->run(Array)
#22 {main}
```

Replicated this error on Omeka 2.0.x and latest Omeka master branch.

**Comment by davidmcclure** [comment]
Looks like this is caused by the addition of the item link at the bottom of the default `item.php` template - https://github.com/scholarslab/Neatline/blame/develop/views/shared/exhibits/item.php#L23-L26. Also, it only breaks when the imports are run as background processes, so I suspect it's some kind of resource-loading issue. I bet it's something related to the set/get_current_item calls. I'll investigate further.

On a related note - anyone know if it's possible to write asynchronous PHPUnit tests, so that we could write real integration test for the jobs? This stuff doesn't get caught because the tests run the jobs synchronously, as regular code inside of a fully-mounted Omeka environment, and so they don't pick up on stuff that _just_ breaks inside background processes, like this. I suppose we could do something like Capybara tests, and check that items get added to exhibits?

**Comment by davidmcclure** [comment]
This is fixed in 2.2.
